### ISSUE TITLE
EventsByTag write batch size: adjust automatically if batch size exceeded

### ISSUE THREAD

**Issue by chbatey** [akka/akka-persistence-cassandra#404]
Cassandra has a max batch size in kbs of the serialized mutation (default 50kb). Setting a default `max-message-batch-size` is hard as it depends on the size of the serialised event. 

Cassandra rejects the write with a warning in the response which could be used to change the batch size dynamically otherwise the writes will never succeed.

**Comment by chbatey** [comment]
Given the write will never succeed until the user changes their `max-message-batch-size` I think we should do this asap

**Comment by patriknw** [comment]
ok, so when we see this exception we automatically decrease the batch size?
I guess that is fine, but making it completely dynamic by trying to increase it again after a while can become more complicated than it's worth. Over time different events may have different sizes.

How does this relate persistAll and atomic writes of several events? We can't split those up.

Apart from persistAll batches may be used for persistAsync.

**Comment by chbatey** [comment]
Yes that would do. Next step could be to add in a check to see if it happens more than once in a given time period to stop a single large event reducing efficiency until restart
